src/Nethermind/Nethermind.Crypto/KzgPolynomialCommitments.cs

```diff
@@ -0,0 +1,65 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.IO;
+using System.Security.Cryptography;
+using System.Threading;
+using Nethermind.Crypto.Properties;
+using Nethermind.Int256;
+
+namespace Nethermind.Crypto
+{
+    public static class KzgPolynomialCommitments
+    {
+        public static readonly UInt256 BlsModulus = UInt256.Parse("73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001", System.Globalization.NumberStyles.HexNumber);
+        public static readonly ulong FieldElementsPerBlob = 4096;
+
+        private const byte KzgBlobHashVersionV1 = 1;
+        private static IntPtr _ckzgSetup = IntPtr.Zero;
+
+        private static readonly ThreadLocal<SHA256> _sha256 = new();
+
+        private static readonly object _inititalizeLock = new();
+        public static void Inititalize()
+        {
+            lock (_inititalizeLock)
+            {
+                if (_ckzgSetup != IntPtr.Zero)
+                {
+                    return;
+                }
+                string tmpFilename = Path.GetTempFileName();
+                using FileStream tmpFileStream = new(tmpFilename, FileMode.OpenOrCreate, FileAccess.Write);
+                using TextWriter tmpFileWriter = new StreamWriter(tmpFileStream);
+                tmpFileWriter.Write(Resources.kzg_trusted_setup);
+                tmpFileWriter.Close();
+                tmpFileStream.Close();
+                _ckzgSetup = Ckzg.Ckzg.LoadTrustedSetup(tmpFilename);
+                File.Delete(tmpFilename);
```

Comment:
Why?

Can't we just make kzg_trusted_setup a EmbeddedResource file? Removng the `Resource` too?